packages/cardano-services/src/Http/HttpServer.ts

```diff
@@ -38,16 +61,38 @@ export class HttpServer extends RunnableModule {
         reviver: (key, value) => (key === '' ? fromSerializableObject(value) : value)
       })
     );
+
     if (this.#config?.metrics?.enabled) {
+      const promRegistry = new Registry();
+      const getServicesHealth = async () => await this.#getServicesHealth();
+      void ((registry) =>
```

Comment:
I think it would be better to assign this to a private class field, then call `reset` on it within `shutdownImpl`

Reply:
Actually, I already did it initially in this way you mentioned (assigning the registry to the private class field, registering the gauge metric to the registry with `this.#registry.registerMetric(metric)` and then calling `this.#registry.clear()` within `shutdownImpl`  which should to `Remove all metrics from the registry`  based on the exposed prom client interface but it turned out didn't work the and metric reference still remained uncleared and the error popped up between test cases: `A metric with the name ${metric.name} has already been registered.` even we have a new instance of HTTP service and reinitialize it in each unit test.

Due to that issue, I dig it into that package and saw there is another way of registering so I chose to approach it this way with `registers: [registry]`.

I could apply back changes with another approach with `registry.registerMetric(metric)` and `clear` it in`shutdownImpl` if needed in order to debug it together but already spent much time on it.

@rhyslbw

Reply:
> assigning the registry to the private class field

Sorry, I mean to assign the _Gauge_ instance to a private field. Rather than call `reset` on shutdown though, I think `set`ting it to `0` is more appropriate

Reply:
Usage is updated in https://github.com/input-output-hk/cardano-js-sdk/pull/403/commits/c79eb0bfd75947dae19a1f1db0a6b9e341fa4b6b, thanks for the help! @rhyslbw